src/sage/combinat/posets/linear_extensions.py

```diff
@@ -252,6 +252,70 @@ def is_greedy(self):
                         return False
         return True
 
+    def is_supergreedy(self):
+        r""""
+        Return ``True`` if the linear extension is supergreedy.
+
+        A linear extension `[e_1, e_2, \ldots, e_n]` is *supergreedy* if
+        for every `i`, either if there is a minimal element `e_{i+1}` in
+        `[e_{i+1}, \ldots, e_n]` which is in the upper cover of `e_j` in
+        `[e_1, \ldots, e_i]` for which j is maximum or if no such element
+        exist `e_{i+1}` is the any of the minimal element in
+        `[e_i+1,\dots, e_n]`.
```

Comment:
I think there is a typo here.

Is this what you wanted to say?
```suggestion
        `[e_1, \ldots, e_i]` for which j is maximum or, if no such element
        exist, `e_{i+1}` is a minimal element in
        `[e_i+1,\dots, e_n]`.
```

Reply:
No, it isn't a typo, I intentionally changed it to imply the definition of supergreedy linear extensions more firmly

Reply:
Oh, I misread your text. Sorry, nevermind.